tests/sentry/workflow_engine/buffer/test_redis_hash_sorted_set_buffer.py

```diff
@@ -138,3 +165,312 @@ def test_delete_keys_bulk(self):
+        result = self.buf.conditional_delete_from_sorted_sets([key], [(999, score)])
+
+        # Should have removed the member
+        assert result[key] == [999]
+
+        # Verify key is empty
+        remaining = self.buf.get_sorted_set(key, 0, time.time() + 10)
+        assert len(remaining) == 0
+
+    def test_conditional_delete_script_loading_failure_propagates(self):
+        """Test that script loading failures are properly propagated."""
+        # Only test script loading failure for cluster/standalone since that's where it's called
+        if not self.buf.is_redis_cluster:
+            pytest.skip("Script loading only happens for RedisCluster")
+
+        # Mock script loading failure
+        with patch.object(self.buf, "_ensure_script_loaded_on_cluster") as mock_ensure:
+            mock_ensure.side_effect = RedisConnectionError("Connection failed")
+
+            with pytest.raises(RedisConnectionError):
+                self.buf.conditional_delete_from_sorted_sets(["key1", "key2"], [(123, 1.0)])
+
+    def test_conditional_delete_slot_based_batching(self):
+        """Test that keys are grouped by slot for efficient batch execution."""
+        if not self.buf.is_redis_cluster:
+            pytest.skip("Slot-based batching is for RedisCluster only")
+
+        # Test slot calculation
+        slot1 = self.buf._calculate_key_slot("test_key_1")
+        slot2 = self.buf._calculate_key_slot("test_key_2")
+        assert isinstance(slot1, int) and 0 <= slot1 < 16384
+        assert isinstance(slot2, int) and 0 <= slot2 < 16384
+
+        # Test key grouping
+        keys = ["key1", "key2", "key3", "key4"]
+        groups = self.buf._group_keys_by_slot(keys)
+
+        # Verify all keys are assigned to groups
+        total_keys = sum(len(group_keys) for group_keys in groups)
+        assert total_keys == len(keys)
+
+        # Test with keys that should have same slot (using hash tags)
+        same_slot_keys = ["prefix{tag}key1", "prefix{tag}key2", "prefix{tag}key3"]
+        same_slot_groups = self.buf._group_keys_by_slot(same_slot_keys)
+
+        # All should be in the same slot due to hash tag
+        assert len(same_slot_groups) == 1
+
+    def test_conditional_delete_functional_slot_batching(self):
+        """Test that slot-based batching works functionally with real data."""
+        if not self.buf.is_redis_cluster:
+            pytest.skip("Slot-based batching is for RedisCluster only")
+
+        # Use hash tags to force keys into same slot for testing
+        keys = ["test{slot}key1", "test{slot}key2", "test{slot}key3"]
+
+        # Add data to all keys
+        for key in keys:
+            self.buf.push_to_sorted_set(key, [100, 200, 300])
+
+        later = time.time() + 1
+
+        # This should execute as a single multi-key script call due to same slot
+        result = self.buf.conditional_delete_from_sorted_sets(keys, [(100, later), (200, later)])
```

Comment:
Similar q, how do we know it executed as one script call? If the rest of the test succeeds?